app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageExternalsManager.kt

```diff
@@ -50,27 +50,40 @@ class NewMessageExternalsManager @Inject constructor() : NewMessageManager() {
     fun observeExternals(arrivedFromExistingDraft: Boolean) = with(newMessageViewModel) {
         Utils.waitInitMediator(initResult, mergedContacts).observe(viewLifecycleOwner) { (_, mergedContacts) ->
             val externalMailFlagEnabled = currentMailbox.externalMailFlagEnabled
+            val trustedDomains = currentMailbox.trustedDomains
             val shouldWarnForExternal = externalMailFlagEnabled && !arrivedFromExistingDraft
             val emailDictionary = mergedContacts.second
             val aliases = currentMailbox.aliases
 
-            updateFields(shouldWarnForExternal, emailDictionary, aliases)
-            updateBanner(shouldWarnForExternal, emailDictionary, aliases)
+            updateFields(shouldWarnForExternal, emailDictionary, aliases, trustedDomains)
+            updateBanner(shouldWarnForExternal, emailDictionary, aliases, trustedDomains)
         }
     }
 
-    private fun updateFields(shouldWarnForExternal: Boolean, emailDictionary: MergedContactDictionary, aliases: List<String>) {
+    private fun updateFields(
+        shouldWarnForExternal: Boolean,
+        emailDictionary: MergedContactDictionary,
+        aliases: List<String>,
+        trustedDomains: List<String>,
+    ) {
         with(binding) {
-            toField.updateExternals(shouldWarnForExternal, emailDictionary, aliases)
-            ccField.updateExternals(shouldWarnForExternal, emailDictionary, aliases)
-            bccField.updateExternals(shouldWarnForExternal, emailDictionary, aliases)
+            toField.updateExternals(shouldWarnForExternal, emailDictionary, aliases, trustedDomains)
+            ccField.updateExternals(shouldWarnForExternal, emailDictionary, aliases, trustedDomains)
+            bccField.updateExternals(shouldWarnForExternal, emailDictionary, aliases, trustedDomains)
         }
     }
 
-    private fun updateBanner(shouldWarnForExternal: Boolean, emailDictionary: MergedContactDictionary, aliases: List<String>) {
+    private fun updateBanner(
+        shouldWarnForExternal: Boolean,
+        emailDictionary: MergedContactDictionary,
+        aliases: List<String>,
+        trustedDomains: List<String>,
```

Comment:
Maybe we could make a dataclass out of these 4 parameters to simplify the 3 functions declarations

Reply:
Finally, out of the last 3. It was better overall.